### PR TITLE
Apidx 884

### PR DESCRIPTION
**Pull request by bubianchi-criteo**
Artifact names and versions and namespace aligned with what defined in the ticket

### CODE REVIEW

**Review by ouvreboite — APPROVED** [review]

File: generator/python/build.gradle [hunk 1]
```diff
@@ -49,39 +50,43 @@ task generateClient(type: GradleBuild) {
...
-                        apiVersion              : parsedJson.info.version,
-                        packageName             : "${packageBase}".toString(),
-                        invokerPackage          : 'Criteo\\SDK',
-                        packageUrl              : 'https://github.com/criteo/criteo-api-python-sdk',
-                        hideGenerationTimestamp : 'true',
-                        artifactId              : "criteo-api-${criteoService}-sdk-${technologyStack}".toString(),
-                        packageVersion          : "${packageVersion}".toString(), // v2021_10.1.211109
                         groupId                 : 'criteo',
-                        version                 : '1.0.0',
+                        apiVersion              : "${artifactVersion}".toString(),
```
Comment (PaulMathon) [inline generator/python/build.gradle:76]: API Version = Artifact verison?

File: generator/python/build.gradle [hunk 2]
```diff
@@ -49,39 +50,43 @@ task generateClient(type: GradleBuild) {
...
+                        artifactId              : "criteo-api-${criteoService}-${versionCategory}-sdk".toString(),
                         description             : "Python SDK for Criteo API ${criteoService} for ${criteoApiVersion} version".toString(),
-                        sourceCompatibility     : '1.8'
+                        hideGenerationTimestamp : 'true',
+                        invokerPackage          : 'Criteo\\SDK',
+                        packageName             : "${namespaceBase}".toString(),
+                        packageUrl              : 'https://github.com/criteo/criteo-api-${language}-sdk',
+                        packageVersion          : "${artifactVersion}".toString(), // v2021_10.1.211109
+                        sourceCompatibility     : '1.8',
+                        version                 : '1.0.0',
```
Comment (PaulMathon) [inline generator/python/build.gradle:85]: What the "version" option is for?

File: generator/php/build.gradle [hunk 3]
```diff
@@ -48,39 +49,45 @@ task generateClient(type: GradleBuild) {
...
                 generateModelTests = false
+                gitUserId = 'criteo'
+                gitRepoId = "criteo-api-${criteoService}-${versionCategory}-sdk".toString()
                 configOptions = [
-                        apiVersion              : parsedJson.info.version,
-                        packageName             : "criteo-api-${criteoService}-sdk-${technologyStack}".toString(),
-                        invokerPackage          : "${packageBase}".toString(),
-                        packageUrl              : 'https://github.com/criteo/criteo-api-php-sdk',
+                        groupId                 : 'criteo',
+                        artifactId              : "criteo-api-${criteoService}-${versionCategory}-sdk".toString(),
```
Comment (PaulMathon) [inline generator/php/build.gradle:76]: We decided not to put "preview" or "stable" in the package name anymore, so I think that it should just be "criteo-api-${criteoService}-sdk"

File: generator/php/build.gradle [hunk 4]
```diff
@@ -48,39 +49,45 @@ task generateClient(type: GradleBuild) {
...
+                        artifactVersion         : "${artifactVersion}".toString(),
+                        description             : "${language.toUpperCase()} SDK for Criteo API ${criteoService} for ${criteoApiVersion} version".toString(),
                         hideGenerationTimestamp : 'true',
-                        artifactId              : "criteo-api-${criteoService}-sdk-${technologyStack}".toString(),
-                        packageVersion          : "${packageVersion}".toString(), // v2021_10.1.211110
-                        groupId                 :  'criteo',
+                        invokerPackage          : "${namespaceBase}".toString(),
+                        packageName             : "criteo-api-${criteoService}-${versionCategory}-sdk".toString(),
+                        packageUrl              : "https://github.com/criteo/criteo-api-${language}-sdk".toString(),
```
Comment (PaulMathon) [inline generator/php/build.gradle:82]: For PHP SDKs there will be 2 repositories, one for each CriteoService:
- [criteo-api-retailmedia-php-sdk](https://github.com/criteo/criteo-api-retailmedia-php-sdk)
- [criteo-api-marketingsolutions-php-sdk](https://github.com/criteo/criteo-api-marketingsolutions-php-sdk)
So I think the string template should more be:
`"https://github.com/criteo/criteo-api-${criteoService}-${language}-sdk".toString()`

File: generator/php/build.gradle [hunk 5]
```diff
@@ -48,39 +49,45 @@ task generateClient(type: GradleBuild) {
...
+                        groupId                 : 'criteo',
+                        artifactId              : "criteo-api-${criteoService}-${versionCategory}-sdk".toString(),
+                        artifactVersion         : "${artifactVersion}".toString(),
+                        description             : "${language.toUpperCase()} SDK for Criteo API ${criteoService} for ${criteoApiVersion} version".toString(),
                         hideGenerationTimestamp : 'true',
-                        artifactId              : "criteo-api-${criteoService}-sdk-${technologyStack}".toString(),
-                        packageVersion          : "${packageVersion}".toString(), // v2021_10.1.211110
-                        groupId                 :  'criteo',
+                        invokerPackage          : "${namespaceBase}".toString(),
+                        packageName             : "criteo-api-${criteoService}-${versionCategory}-sdk".toString(),
```
Comment (PaulMathon) [inline generator/php/build.gradle:81]: Same as for the artifactId

File: generator/java/build.gradle [hunk 6]
```diff
@@ -64,27 +69,26 @@ task generateClient(type: GradleBuild) {
...
-                    apiVersion               : parsedJson.info.version,  // 2021-10
-                    artifactDescription      : "Java SDK for Criteo API ${criteoService} for ${criteoApiVersion} version".toString(),
-                    artifactUrl              : 'https://github.com/criteo/criteo-api-java-sdk',
-                    modelPackage             : "${packageBase}.model".toString(),
-                    scmConnection            : 'scm:git:git://github.com/criteo/criteo-api-java-sdk.git',
-                    scmDeveloperConnection   : 'scm:git:ssh://github.com:criteo/criteo-api-java-sdk.git',
-                    scmUrl                   : 'https://github.com/criteo/criteo-api-java-sdk',
+                    apiPackage               : "${namespaceBase}.api".toString(), // com.criteo.api.marketingsolutions.v2021_10.api
+                    artifactDescription      : "${language.toUpperCase()} SDK for Criteo API ${criteoService} for ${criteoApiVersion} version".toString(),
+                    artifactId               : "criteo-api-${criteoService}-${versionCategory}-sdk".toString(), // criteo-api-marketingsolutions-preview-sdk
```
Comment (PaulMathon) [inline generator/java/build.gradle:74]: I think we decided that the versionCategory should not be included in the package name right?

File: generator/python/build.gradle [hunk 7]
```diff
@@ -49,39 +50,43 @@ task generateClient(type: GradleBuild) {
...
-                        packageName             : "${packageBase}".toString(),
-                        invokerPackage          : 'Criteo\\SDK',
-                        packageUrl              : 'https://github.com/criteo/criteo-api-python-sdk',
-                        hideGenerationTimestamp : 'true',
-                        artifactId              : "criteo-api-${criteoService}-sdk-${technologyStack}".toString(),
-                        packageVersion          : "${packageVersion}".toString(), // v2021_10.1.211109
                         groupId                 : 'criteo',
-                        version                 : '1.0.0',
+                        apiVersion              : "${artifactVersion}".toString(),
+                        artifactId              : "criteo-api-${criteoService}-${versionCategory}-sdk".toString(),
```
Comment (PaulMathon) [inline generator/python/build.gradle:77]: same as before, the version category should not be included in the package name

File: generator/php/build.gradle (hunk 3 above)
Comment (bubianchi-criteo) [reply generator/php/build.gradle]: Let's update the story accordingly, and take into account these latest decisions

File: generator/php/build.gradle (hunk 4 above)
Comment (bubianchi-criteo) [reply generator/php/build.gradle]: We will update the story and take these latest decision into account in next commit

File: generator/java/build.gradle (hunk 6 above)
Comment (bubianchi-criteo) [reply generator/java/build.gradle]: We will update the story and take these latest decision into account in next commit

File: generator/python/build.gradle (hunk 7 above)
Comment (bubianchi-criteo) [reply generator/python/build.gradle]: We will update the story and take these latest decision into account in next commit

File: generator/python/build.gradle (hunk 2 above)
Comment (bubianchi-criteo) [reply generator/python/build.gradle]: From the open api tools generator source code, "version" is used as the "API version", whereas in the default mustache, we have: 
in java: 
 - in pom, "package version" = artifactVersion
 - in readme, "API version" = appVersion ; "project version"= artifactVersion
 
in python : 
- in configuration , "Version of the API" =  version;  "SDK Package Version" = packageVersion 
- in readme, "API version" = appVersion
- in setup, "VERSION" = packageVersion

in php: 
- in composer, "version" = artifactVersion
- in readme, "API version " = appVersion,  "Package version" = artifactVersion

appVersion is retrieved from the version in the swagger (it seems not to be overridable)

File: generator/python/build.gradle (hunk 1 above)
Comment (bubianchi-criteo) [reply generator/python/build.gradle:76]: see next comment

File: generator/php/build.gradle (hunk 5 above)
Comment (bubianchi-criteo) [reply generator/php/build.gradle]: see comment on "version"

**Review by arturrudiuk-criteo — APPROVED** [review]

File: generator/java/build.gradle [hunk 8]
```diff
@@ -45,17 +46,21 @@ task generateClient(type: GradleBuild) {
         
         def clientOutputDirPerVersion = clientOutputDir + "/" + parsedApiName;
         def specPath = it.path;
-        def packageBase =  criteoApiVersion; 
-        if (packageBase.matches("[0-9].*"))
-             packageBase= "v" + packageBase;
-
-        packageBase = "${criteoPackage}.${criteoService}.${packageBase}".replace('.', nameSeparator).toString(); 
-        def packageVersion = "${criteoApiVersion}.${generatorVersion}.${getFormattedDate()}"; // example : 2021_10.1.211110
+        def versionCategory = criteoApiVersion == 'preview' ? 'preview' : 'stable'
```
Comment (PaulMathon) [inline generator/java/build.gradle:49]: Maybe we could replace versionCategory by "isPreviewVersion"?

File: generator/java/build.gradle [hunk 9]
```diff
@@ -64,27 +69,26 @@ task generateClient(type: GradleBuild) {
...
-                    artifactDescription      : "Java SDK for Criteo API ${criteoService} for ${criteoApiVersion} version".toString(),
-                    artifactUrl              : 'https://github.com/criteo/criteo-api-java-sdk',
-                    modelPackage             : "${packageBase}.model".toString(),
-                    scmConnection            : 'scm:git:git://github.com/criteo/criteo-api-java-sdk.git',
-                    scmDeveloperConnection   : 'scm:git:ssh://github.com:criteo/criteo-api-java-sdk.git',
-                    scmUrl                   : 'https://github.com/criteo/criteo-api-java-sdk',
+                    apiPackage               : "${namespaceBase}.api".toString(), // com.criteo.api.marketingsolutions.v2021_10.api
+                    artifactDescription      : "${language.toUpperCase()} SDK for Criteo API ${criteoService} for ${criteoApiVersion} version".toString(),
+                    artifactId               : "criteo-api-${criteoService}-sdk".toString(), // criteo-api-marketingsolutions-preview-sdk
+                    artifactVersion          : "${artifactVersion}".toString(), // 2021.10.1.211110
```
Comment (PaulMathon) [inline generator/java/build.gradle:75]: Update comments please 😄 
// criteo-api-marketingsolutions-sdk
// 2021.10.**0**.211110

File: generator/php/resources/templates/README.mustache [hunk 10]
```diff
@@ -8,10 +8,7 @@ More information: [https://developers.criteo.com/marketing-solutions/docs/php-ap
 
 This PHP package is automatically generated by the [OpenAPI Generator](https://openapi-generator.tech) project:
 
-- API version: {{apiVersion}}
-{{#apiVersion}}
-- Package version: {{apiVersion}}
-{{/apiVersion}}
+- API version: {{artifactVersion}}
```
Comment (PaulMathon) [inline generator/php/resources/templates/README.mustache:11]: the artifactVerison corresponds more to the version of the package right?

File: generator/python/build.gradle [hunk 11]
```diff
@@ -49,39 +50,41 @@ task generateClient(type: GradleBuild) {
...
+  
+        def versionInNamespace =  criteoApiVersion
+        if (versionInNamespace.matches("[0-9].*"))
+             versionInNamespace= "v" + versionInNamespace
 
-        packageBase = "${criteoPackage}.${criteoService}.${packageBase}".replace('.', nameSeparator).toString(); // example : criteo_api_marketingsolutions_v2021_10
-        def packageVersion = "${criteoApiVersion}.${generatorVersion}.${getFormattedDate()}"; // example : 2021_10.1.211110
-
-        def generateTask = task("openApiGenerate_python_" + parsedApiName, type: GenerateTask.class) {
+        def namespaceBase = "${criteoPackage}.${criteoService}.${versionInNamespace}".replace('.', nameSeparator).toString(); // example: criteo\api\marketingsolutions\v2021_10
```
Comment (PaulMathon) [inline generator/python/build.gradle:59]: The comment seems wrong, should be criteo_api_marketingsolutions_v2021_10 right?

File: generator/python/build.gradle [hunk 12]
```diff
@@ -49,39 +50,41 @@ task generateClient(type: GradleBuild) {
...
-                        packageVersion          : "${packageVersion}".toString(), // v2021_10.1.211109
                         groupId                 : 'criteo',
-                        version                 : '1.0.0',
-                        description             : "Python SDK for Criteo API ${criteoService} for ${criteoApiVersion} version".toString(),
-                        sourceCompatibility     : '1.8'
+                        apiVersion              : "${artifactVersion}".toString(),
+                        artifactId              : "criteo-api-${criteoService}-sdk".toString(),
+                        description             : "${language.toUpperCase()} SDK for Criteo API ${criteoService} for ${criteoApiVersion} version".toString(),
+                        hideGenerationTimestamp : 'true',
+                        invokerPackage          : 'Criteo\\SDK',
```
Comment (PaulMathon) [inline generator/python/build.gradle:80]: shouldn't the invokerPackage be namespaceBase?

File: generator/java/build.gradle (hunk 8 above)
Comment (PaulMathon) [reply generator/java/build.gradle:49]: Just a question, where does the criteoApiVersion variable come from?

File: generator/python/build.gradle [hunk 13]
```diff
@@ -49,39 +50,43 @@ task generateClient(type: GradleBuild) {
...
                 templateDir = "$resourcesDir/templates/".toString()
                 inputSpec = specPath
                 outputDir = clientOutputDirPerVersion
                 removeOperationIdPrefix = true
                 generateApiTests = false
                 generateModelTests = false
-                gitUserId='criteo'
-                gitRepoId='criteo-api-python-sdk'
+                gitUserId = 'criteo'
+                gitRepoId = "criteo-api-${criteoService}-${versionCategory}-sdk".toString()
```
[the remark this answers is not in the document]
Comment (bubianchi-criteo) [reply generator/python/build.gradle]: Done in next commit

File: generator/java/build.gradle (hunk 8 above)
Comment (bubianchi-criteo) [reply generator/java/build.gradle:49]: Done

File: generator/java/build.gradle (hunk 9 above)
Comment (bubianchi-criteo) [reply generator/java/build.gradle:75]: Done

File: generator/php/resources/templates/README.mustache (hunk 10 above)
Comment (bubianchi-criteo) [reply generator/php/resources/templates/README.mustache:11]: "-Package version : {{artifactVersion}}"

File: generator/python/build.gradle (hunk 11 above)
Comment (bubianchi-criteo) [reply generator/python/build.gradle:59]: Done

File: generator/python/build.gradle (hunk 12 above)
Comment (bubianchi-criteo) [reply generator/python/build.gradle:80]: invokerPackage doesn't seem to have any impact on generated files. Let's keep this potential clean up for another review.

File: generator/python/build.gradle [hunk 14]
```diff
@@ -49,39 +50,39 @@ task generateClient(type: GradleBuild) {
...
-                        description             : "Python SDK for Criteo API ${criteoService} for ${criteoApiVersion} version".toString(),
-                        sourceCompatibility     : '1.8'
+                        apiVersion              : "${artifactVersion}".toString(),
+                        artifactId              : "criteo-api-${criteoService}-sdk".toString(),
+                        description             : "${language.toUpperCase()} SDK for Criteo API ${criteoService} for ${criteoApiVersion} version".toString(),
+                        hideGenerationTimestamp : 'true',
+                        invokerPackage          : 'Criteo\\SDK',
+                        packageName             : "${namespaceBase}".toString(),
+                        packageUrl              : "https://github.com/criteo/criteo-api-${language}-sdk".toString(),
+                        packageVersion          : "${artifactVersion}".toString(), // v2021_10.1.211109
```
Comment (PaulMathon) [inline generator/python/build.gradle:81]: For another review: update example in comment

**Review by PaulMathon — APPROVED** [review]

**Pull request merged**